storage/indexer/src/db_tailer.rs

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    schema::{
+        event_by_key::EventByKeySchema, event_by_version::EventByVersionSchema,
+        indexer_metadata::TailerMetadataSchema, transaction_by_account::TransactionByAccountSchema,
+    },
+    utils::{
+        error_if_too_many_requested, get_first_seq_num_and_limit, AccountTransactionVersionIter,
+        MAX_REQUEST_LIMIT,
+    },
+};
+use aptos_config::config::index_db_tailer_config::IndexDBTailerConfig;
+use aptos_schemadb::{ReadOptions, SchemaBatch, DB};
+use aptos_storage_interface::{
+    db_ensure as ensure, db_other_bail as bail, AptosDbError, DbReader, Result,
+};
+use aptos_types::{
+    account_address::AccountAddress,
+    contract_event::{ContractEvent, EventWithVersion},
+    event::EventKey,
+    indexer::db_tailer_reader::Order,
+    transaction::{AccountTransactionsWithProof, Version},
+};
+use std::sync::Arc;
+
+pub struct DBTailer {
```

Comment:
why not just call it InternalIndexer or something along that line? 
`DBTrailer::db` sounds like what the thing is tailing.

Reply:
I feel it is just tailing the db

Reply:
renamed to DBindexer